src/apps/api/tests/test_competitions.py

```diff
@@ -138,3 +142,125 @@ def test_manual_migration_makes_submissions_out_of_only_parents_not_children(sel
+        response = c.get(f'/api/competitions/{self.comp.id}/results.json?title={leaderboard_choice}', HTTP_ACCEPT='application/json')
+        self.assertEqual(response.status_code, 200)
+        content = json.loads(response.content)
+
+        for title in content.keys():
+            assert leaderboard_choice in title
+
+    def test_get_competition_leaderboard_by_id_as_json(self):
+        # Make sure when getting leaderboard by id you get exactly one leaderboard with matching title
+        c = Client()
+        c.login(username="creator2", password="creator2")
+        leaderboard_choice = random.choice(list(self.leaderboard_ids_to_titles.keys()))
+        response = c.get(f'/api/competitions/{self.comp.id}/results.json?id={leaderboard_choice}', HTTP_ACCEPT='application/json')
+        self.assertEqual(response.status_code, 200)
+        content = json.loads(response.content)
+
+        response_title = list(content.keys())
+        assert len(response_title) == 1
+        assert response_title[0] == f'{self.leaderboard_ids_to_titles[leaderboard_choice]}({leaderboard_choice})'
+
+    def test_get_competition_leaderboard_by_id_as_csv(self):
+        c = Client()
+        c.login(username="creator2", password="creator2")
+        leaderboard_choice = random.choice(list(self.leaderboard_ids_to_titles.keys()))
+        response = c.get(f'/api/competitions/{self.comp.id}/results.csv?id={leaderboard_choice}', HTTP_ACCEPT='text/csv')
+        self.assertEqual(response.status_code, 200)
+
+        content = response.content.decode('utf-8')
+        csv_reader = csv.reader(StringIO(content))
+        csv_header = list(csv_reader)[0]
+        csv_header.pop(0)
+
+        for column_title in self.leaderboard_ids_to_columns[leaderboard_choice]:
+            assert f'{column_title}({self.leaderboard_ids_to_columns[leaderboard_choice][column_title]})'
+
+    def test_get_competition_leaderboard_by_title_as_csv(self):
+        c = Client()
+        c.login(username="creator2", password="creator2")
+        leaderboard_choice = random.choice(list(self.leaderboard_ids_to_titles.keys()))
+        response = c.get(f'/api/competitions/{self.comp.id}/results.csv?title={self.leaderboard_ids_to_titles[leaderboard_choice]}({leaderboard_choice})', HTTP_ACCEPT='text/csv')
+        self.assertEqual(response.status_code, 200)
+
+        content = response.content.decode('utf-8')
+        csv_reader = csv.reader(StringIO(content))
+        csv_header = list(csv_reader)[0]
+        csv_header.pop(0)
+
+        for column_title in self.leaderboard_ids_to_columns[leaderboard_choice]:
+            assert f'{column_title}({self.leaderboard_ids_to_columns[leaderboard_choice][column_title]})'
+
+    def test_get_competition_leaderboard_as_zip(self):
+        c = Client()
+        c.login(username="creator2", password="creator2")
+        response = c.get(f'/api/competitions/{self.comp.id}/results.zip')
```

Comment:
should use `reverse()` to build URLs instead of manually putting them in, that makes them a bit easier to maintain in the long run

Reply:
Done!